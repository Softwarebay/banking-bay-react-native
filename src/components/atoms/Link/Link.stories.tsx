import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';

import { ViewStory } from 'utils/storybook/ViewStory';

import { Link } from './Link';

storiesOf('Atoms/Link', module)
	.addDecorator((getStory) => <ViewStory center>{getStory()}</ViewStory>)
	.add('Default', () => (
		<Link
			testID={text('testID', 'Link-test-id')}
			text={text('text', 'Click me!')}
			onPress={action('onPress')}
		/>
	));
