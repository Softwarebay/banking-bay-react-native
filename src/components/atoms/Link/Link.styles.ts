import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	link: {
		color: colors.pink,
		fontFamily: 'MulishMedium',
		fontSize: 16,
	},
	pressed: {
		opacity: 0.8,
	},
});
