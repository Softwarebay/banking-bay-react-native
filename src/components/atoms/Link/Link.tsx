import { Pressable, StyleProp, Text, TextStyle } from 'react-native';

import { styles } from './Link.styles';

export interface Props {
	text: string;
	onPress: () => void;
	testID: string;
	style?: StyleProp<TextStyle>;
}

export const Link = ({ text, onPress, testID, style }: Props) => {
	return (
		<Pressable
			hitSlop={20}
			style={({ pressed }) => pressed && styles.pressed}
			testID={testID}
			onPress={onPress}
		>
			<Text style={[styles.link, style]}>{text}</Text>
		</Pressable>
	);
};
