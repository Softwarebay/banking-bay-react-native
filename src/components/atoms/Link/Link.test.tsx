import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Link } from './Link';

import type { Props } from './Link';

describe('Link', () => {
	const mockOnPress = jest.fn<Props['onPress'], []>();
	const props: Props = {
		onPress: mockOnPress,
		text: 'Link',
		testID: 'link-test-id',
		style: {
			marginBottom: 20,
		},
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<Link {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render text passed via props', () => {
		render(<Link {...props} />);

		const link = screen.getByTestId('link-test-id');

		expect(link).toHaveTextContent('Link');
	});

	it('should apply additional text styles from props', () => {
		render(<Link {...props} />);

		const linkText = screen.getByText('Link');

		expect(linkText).toHaveStyle({
			marginBottom: 20,
		});
	});

	it('should handle press events', () => {
		render(<Link {...props} />);

		const link = screen.getByTestId('link-test-id');

		fireEvent.press(link);

		expect(mockOnPress).toHaveBeenCalledTimes(1);
	});
});
