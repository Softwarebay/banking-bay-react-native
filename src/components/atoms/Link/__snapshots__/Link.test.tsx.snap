// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={20}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={false}
  testID="link-test-id"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FF5887",
          "fontFamily": "MulishMedium",
          "fontSize": 17,
          "lineHeight": 27.2,
        },
        Object {
          "marginBottom": 20,
        },
      ]
    }
  >
    Link
  </Text>
</View>
`;
